cub/cub/device/device_transform.cuh

```diff
@@ -0,0 +1,230 @@
+  template <typename... RandomAccessIteratorsIn, typename RandomAccessIteratorOut, typename TransformOp>
+  CUB_RUNTIME_FUNCTION static cudaError_t Transform(
+    int count,
+    ::cuda::std::tuple<RandomAccessIteratorsIn...> inputs,
+    RandomAccessIteratorOut output,
+    TransformOp transform_op,
+    cudaStream_t stream = nullptr)
+  {
+    CUB_DETAIL_NVTX_RANGE_SCOPE("cub::DeviceTransform::Transform");
+    return detail::transform::
+      dispatch_t<false, int, ::cuda::std::tuple<RandomAccessIteratorsIn...>, RandomAccessIteratorOut, TransformOp>::
+        dispatch(count, ::cuda::std::move(inputs), ::cuda::std::move(output), ::cuda::std::move(transform_op), stream);
+  }
+
+  // temp storage overload
+  template <typename... RandomAccessIteratorsIn, typename RandomAccessIteratorOut, typename TransformOp>
+  CUB_RUNTIME_FUNCTION static cudaError_t Transform(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    int count,
+    ::cuda::std::tuple<RandomAccessIteratorsIn...> inputs,
+    RandomAccessIteratorOut output,
+    TransformOp transform_op,
+    cudaStream_t stream = nullptr)
+  {
+    if (d_temp_storage == nullptr)
+    {
+      temp_storage_bytes = 1;
+      return cudaSuccess;
+    }
+
+    return Transform(
+      count, ::cuda::std::move(inputs), ::cuda::std::move(output), ::cuda::std::move(transform_op), stream);
+  }
+
+  // One input stream, one output stream
+
+  template <typename RandomAccessIteratorIn, typename RandomAccessIteratorOut, typename TransformOp>
+  CUB_RUNTIME_FUNCTION static cudaError_t Transform(
+    int count,
+    RandomAccessIteratorIn input,
+    RandomAccessIteratorOut output,
+    TransformOp transform_op,
+    cudaStream_t stream = nullptr)
+  {
+    return Transform(
+      count,
+      ::cuda::std::make_tuple(::cuda::std::move(input)),
+      ::cuda::std::move(output),
+      ::cuda::std::move(transform_op),
+      stream);
+  }
+
+  // temp storage overload
+  template <typename RandomAccessIteratorIn, typename RandomAccessIteratorOut, typename TransformOp>
+  CUB_RUNTIME_FUNCTION static cudaError_t Transform(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    int count,
+    RandomAccessIteratorIn input,
+    RandomAccessIteratorOut output,
+    TransformOp transform_op,
+    cudaStream_t stream = nullptr)
+  {
+    if (d_temp_storage == nullptr)
+    {
+      temp_storage_bytes = 1;
+      return cudaSuccess;
+    }
+
+    return Transform(
+      count,
+      ::cuda::std::make_tuple(::cuda::std::move(input)),
+      ::cuda::std::move(output),
+      ::cuda::std::move(transform_op),
+      stream);
+  }
+
+  // Many input streams, one output stream, address stable
+
+  /// @rst
+  /// Overview
+  /// +++++++++++++++++++++++++++++++++++++++++++++
+  /// Like \ref cub::DeviceTransform::Transform, but the objects passed to the call operator of the transformation
+  /// operation are guaranteed to reside in the input streams and are never copied.
```

Comment:
question: I'm afraid that stream is overloaded term here. Can we explain this concept without mentioning it?

Reply:
Replaced by `sequence`.